org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/RefactorProposalUtility.java

```diff
@@ -663,6 +666,26 @@ public static CUCorrectionProposal getExtractMethodProposal(CodeActionParams par
 		return null;
 	}
 
+	public static CUCorrectionProposal getIntroduceParameterRefactoringProposals(CodeActionParams params, IInvocationContext context, ASTNode coveringNode, boolean returnAsCommand, IProblemLocationCore[] problemLocations)
+			throws CoreException {
+		final ICompilationUnit cu = context.getCompilationUnit();
+		final IntroduceParameterRefactoring introduceParameterRefactoring = new IntroduceParameterRefactoring(cu, context.getSelectionOffset(), context.getSelectionLength());
+		LinkedProposalModelCore linkedProposalModel = new LinkedProposalModelCore();
+		introduceParameterRefactoring.setLinkedProposalModel(linkedProposalModel);
+		if (introduceParameterRefactoring.checkInitialConditions(new NullProgressMonitor()).isOK()) {
+			introduceParameterRefactoring.setParameterName(introduceParameterRefactoring.guessedParameterName());
+			String label = RefactoringCoreMessages.IntroduceParameterRefactoring_name + "...";
+			int relevance = (problemLocations != null && problemLocations.length > 0) ? IProposalRelevance.EXTRACT_CONSTANT_ERROR : IProposalRelevance.EXTRACT_CONSTANT;
+			if (returnAsCommand) {
```

Comment:
that seems unnecessary

Reply:
See the previous comment and some existing examples:
- https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/RefactorProposalUtility.java#L302
- https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/RefactorProposalUtility.java#L440